src/bridge/cockpitpeer.c

```diff
@@ -366,12 +378,15 @@ on_other_control (CockpitTransport *transport,
             {
               gchar *user_hex;
               gchar *user;
+              gchar *stderr;
               self->startup_auth_cookie = g_strdup (cookie);
               cockpit_authorize_subject (challenge, &user_hex);
               user = cockpit_hex_decode (user_hex, -1, NULL);
-              cockpit_router_prompt (self->router, user, prompt, on_answer, self);
+              stderr = startup_take_stderr (self);
```

Comment:
This is pretty extremely pedantic, but it might happen that sudo prints some output to stderr, then calls askpass, which prints an auth control message to stdout.  On a busy system it might end up that cockpit-bridge doesn't get a change to run again until after that all happens (maybe it was in swap?) and then when it comes back it has a 50/50 chance to read from the socket or the stderr pipe first.  If it reads from the socket first, we'll see the control message before we have all of our stderr data from sudo.

We could fix this by manually polling stderr to check if there's outstanding data before we empty the buffer.  That's probably something that belongs in `CockpitPipe`.

Addressing this now, however unlikely we expect it to be in practice, might avoid some future flakes turning up because of this.  stdout/err ordering issues when they are not on the same fd/pty/whatever are really tricky to trace down.

Reply:
Good point!

Reply:
I have added the `drain_error` function to read all that is available when getting the stderr output.